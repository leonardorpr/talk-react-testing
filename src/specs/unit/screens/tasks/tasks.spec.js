import React from 'react'
import { render } from 'react-testing-library'

import { Provider } from 'react-redux'

import { mockTasks, mockStore } from 'specs/helpers'

import Tasks, { handleTasks } from 'screens/tasks'

describe('[SCREENS] Tasks', () => {
  it('Should render correctly', () => {
    const { container } = render(
      <Provider store={mockStore({ tasks: { tasks: [] } })}>
        <Tasks createTask={jest.fn()} toggleTask={jest.fn()} />
      </Provider>
    )

    expect(container).toMatchSnapshot()
  })

  it('Should only return completed tasks', () => {
    const tasks = handleTasks(mockTasks, 'done')

    expect(tasks).toHaveLength(1)
  })

  it('Should only return tasks to do', () => {
    const tasks = handleTasks(mockTasks, 'todo')

    expect(tasks).toHaveLength(2)
  })
})
