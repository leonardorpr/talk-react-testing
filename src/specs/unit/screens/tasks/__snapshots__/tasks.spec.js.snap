// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SCREENS] Tasks Should render correctly 1`] = `
<div>
  <div
    class="sc-VigVT jYaxIa"
  >
    <form
      class="sc-dnqmqq ewbNOT"
    >
      <div
        class="sc-iwsKbI cFpGCb"
      >
        <input
          class="sc-gZMcBi jwxFqm"
          placeholder="Task name"
          type="text"
          value=""
        />
        <input
          class="sc-gZMcBi jwxFqm"
          placeholder="Estimated time"
          type="text"
          value=""
        />
        <button
          class="sc-gqjmRU foOVYb"
          type="submit"
        >
          Create
        </button>
      </div>
    </form>
    <div
      class="sc-jTzLTM kZAMhy"
    >
      <div
        class="sc-bdVaJa huxbce"
      >
        <div
          class="sc-bwzfXH ffXQkl"
        >
          <h3
            class="sc-htpNat boHzRd"
          >
            To Do
          </h3>
        </div>
      </div>
      <div
        class="sc-bdVaJa huxbce"
      >
        <div
          class="sc-bwzfXH ffXQkl"
        >
          <h3
            class="sc-htpNat boHzRd"
          >
            Done
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;
